### DISCUSSION TITLE
Design Problem Definition

### DISCUSSION THREAD

**Discussion by yonghoonlee** [WEIS-UIUC-UofM/WEIS#2] [Show and tell]
@elefer3 
I am trying to understand [this part](https://github.com/WEIS-UIUC-UofM/WEIS/blob/df419826478a66b31f76131d2d45134e8ea437b3/weis/optimization/DC_comp.py#L859) of the code, and for me, it seems to be very specific code for certain wind turbine type and certain DOE variables. Is there any better way to extract variable names from the SMT object and use them as variables for optimization? Let's discuss a bit further on how we can make work better.

**Comment by elefer3** [comment]
@yonghoonlee that part of the code is temporary, In practice that dictionary (DCA_req) would come from Saeid's code, specifying what are the design variables, objective etc. I have those values as a placeholder so that I can run the optimization until my code interfaces with Saeid's.
To clarify more, the variable names from the SMT object are extracted in line 29 (def read_sm(self, sm_file, modeling_options, opt_options)), but the code doesn't assume that all the design variables in the STM object are to be used in the optimization.

**Reply by yonghoonlee** [reply]
Thank you for clarification!
